pkg/app/api/grpcapi/piped_api.go

```diff
@@ -963,6 +966,129 @@ func (a *PipedAPI) ReportUnregisteredApplicationConfigurations(ctx context.Conte
 	return nil, status.Errorf(codes.Unimplemented, "ReportUnregisteredApplicationConfigurations is not implemented yet")
 }
 
+// CreateDeploymentChain creates a new deployment chain object and all required commands to
+// trigger deployment for applications in the chain.
+func (a *PipedAPI) CreateDeploymentChain(ctx context.Context, req *pipedservice.CreateDeploymentChainRequest) (*pipedservice.CreateDeploymentChainResponse, error) {
+	projectID, _, _, err := rpcauth.ExtractPipedToken(ctx)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Add a check to ensure that the application of the requested deployment belongs to that this Piped or not.

Reply:
addressed by https://github.com/pipe-cd/pipe/pull/2815/commits/91a442b023c4f1a19c95b4cdc36c5c35d8c3b0c9